webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -1608,17 +1620,92 @@ public static ByteArrayOutputStream requestsToZip(ServletContext context, String
         zipOut.putNextEntry(null, parameters);
         // Note that we're not passing the password into this function. It seems pretty silly
         // to put a password protected pdf inside of a password protected zip file.
-        ByteArrayOutputStream baos = requestsToPdf(globalTitle, generationDate, scrambleRequests, null);
+
+        boolean ordered = false;
+        ByteArrayOutputStream baos = requestsToPdf(globalTitle, generationDate, scrambleRequests, null, ordered);
         zipOut.write(baos.toByteArray());
         zipOut.closeEntry();
 
+        ordered = true;
+        for (ScrambleRequest scrambleRequest : scrambleRequests) { // Check if the schedule is available. This will also let legacy as is.
+            if (scrambleRequest.roundStartTime == null || scrambleRequest.roomNames == null || scrambleRequest.timeZone == null) {
+                ordered = false;
+                break;
+            }
+        }
+        if (ordered) {
+
+            // we split scrambleRequest based on day and room
+            ArrayList<Integer> dateList = new ArrayList<Integer>();
+            ArrayList<ArrayList<String>> roomList = new ArrayList<ArrayList<String>>();
+            ArrayList<ArrayList<ArrayList<ScrambleRequest>>> schedule = new ArrayList<ArrayList<ArrayList<ScrambleRequest>>>();
+
+            // let's say attempt = 3 > 1. we are generating 3x more scrambles than needed.
+            for (ScrambleRequest scrambleRequest : scrambleRequests) {
+
+                int attempts = scrambleRequest.scrambles.length;
+//              System.out.println("Attempts: "+attempts);
+
+                Calendar cal = Calendar.getInstance(TimeZone.getTimeZone(scrambleRequest.timeZone));
+                cal.setTime(scrambleRequest.roundStartTime);
+
+                // this will work as long as WCA forbid competitions between years overlap
```

Comment:
FYI there has been "multiyear" competition already, see [Welcome to 2019](https://www.worldcubeassociation.org/competitions/Welcometo2019).

Reply:
[Fixed](https://github.com/campos20/tnoodle/commit/e721f21dfefa9ea1247dd2cb3095edcda14c99e4#diff-5c6b26a54fa9a1a113fc535c6c81c91cR1654), thanks!